src/renderer/__tests__/App.test.tsx

```diff
@@ -0,0 +1,17 @@
+import '@testing-library/jest-dom';
+import { render } from '@testing-library/react';
+
+import App from '../App';
+
+describe('App', () => {
+  /**
+   * We test StoreChangeObserver separately from the rest of the App component as it does some async logic
+   * (loading recent projects from IPC). This can be worked around with 'waitFor' but unfortunately MUI's Modal
+   * component breaks when we do that. So for now we test separately
+   */
+  it('should render successfully without wait when StoreChangeObserver is disabled', async () => {
+    const rendered = render(<App hasStoreChangeObserver={false} />);
+
+    expect(rendered).toBeTruthy();
```

Comment:
Not a critique but a question: How does toBeTruthy() work?

Reply:
It converts the value to a boolean and returns whether or not the result is true

e.g.

"hello", 6, true, [1,2,3], { key: "val" } all become true

and

"", [], 0, null, false, undefined all become false

So what it's checking is if the app render produces something meaningful other than just null/undefined basically

Reply:
https://developer.mozilla.org/en-US/docs/Glossary/Truthy

truthiness is a gross javascript thing that causes lots of problems haha